packages/@react-spectrum/s2/src/Accordion.tsx

```diff
@@ -0,0 +1,429 @@
+import React, {createContext, forwardRef, ReactElement, useContext} from 'react';
+import {size as sizeValue, style} from '../style/spectrum-theme' with { type: 'macro' };
+import {StyleString} from '../style/types';
+import {useDOMRef} from '@react-spectrum/utils';
+import {useSpectrumContextProps} from './useSpectrumContextProps';
+
+
+export interface AccordionGroupProps extends UnsafeStyles, DOMProps, SlotProps {
+  /** The accordion items in the accordion group. */
+  children: React.ReactNode,
+  /** Spectrum-defined styles, returned by the `style()` macro. */
+  styles?: StyleString,
+  /**
+   * The size of the accordion.
+   * @default "M"
+   */
+  size?: 'S' | 'M' | 'L' | 'XL',
+  /**
+   * The amount of space between the accordion items.
+   * @default "regular"
+   */
+  density?: 'compact' | 'regular' | 'spacious',
+  /** Whether the accordion should be displayed with a quiet style. */
+  isQuiet?: boolean,
+  /** Whether the accordion should be disabled. */
+  isDisabled?: boolean
+}
+
+const groupStyles = style({
+  display: 'flex',
+  flexDirection: 'column'
+}, getAllowedOverrides());
+
+export const AccordionGroupContext = createContext<ContextValue<AccordionGroupProps, DOMRefValue<HTMLDivElement>>>(null);
+
+let InternalAccordionGroupContext = createContext<Pick<AccordionGroupProps, 'size'| 'isQuiet'| 'density' | 'isDisabled'>>({});
+
+function AccordionGroup(props: AccordionGroupProps, ref: DOMRef<HTMLDivElement>) {
+  [props, ref] = useSpectrumContextProps(props, ref, AccordionGroupContext);
+  let domRef = useDOMRef(ref);
+  let {
+    UNSAFE_style,
+    UNSAFE_className = '',
+    size = 'M',
+    density = 'regular',
+    isQuiet,
+    isDisabled,
+    ...otherProps
+  } = props;
+  const domProps = filterDOMProps(otherProps);
+  return (
+    <Provider
+      values={[
+        [InternalAccordionGroupContext, {size, isQuiet, density, isDisabled}]
+      ]}>
+      <div
+        {...domProps}
+        ref={domRef}
+        style={UNSAFE_style}
+        className={(UNSAFE_className ?? '') + groupStyles(null, props.styles)}>
+        {props.children}
+      </div>
+    </Provider>
+  );
+}
+
+/**
+ * A accordion group is a container for multiple accordion items.
+ */
+let _AccordionGroup = /*#__PURE__*/ (forwardRef as forwardRefType)(AccordionGroup);
+export {_AccordionGroup as AccordionGroup};
+
+export interface AccordionItemProps extends RACAccordionItemProps, UnsafeStyles, DOMProps {
+  /**
+   * The size of the accordion item.
+   * @default "M"
+   */
+  size?: 'S' | 'M' | 'L' | 'XL',
+  /**
+   * The amount of space between the accordion items.
+   * @default "regular"
+   */
+  density?: 'compact' | 'regular' | 'spacious',
+  /** Whether the accordion item should be displayed with a quiet style. */
+  isQuiet?: boolean,
+  /** The contents of the accordion item, consisting of an AccordionHeader and AccordionPanel. */
+  children: [ReactElement<AccordionHeaderProps>, ReactElement<AccordionPanelProps>],
```

Comment:
React 16/17/18 don't support this, not sure about 19, but I didn't think we can support enforcing this at the TS level. Which probably means we can't enforce this at all, other than visually, which isn't that helpful. Maybe it doesn't matter the order